### PR TITLE
Implemented a list of all active sessions

### PR DESCRIPTION
**Pull request by zeroknots**
In order to allow up to wipe all existing sessions, I have ported over EnumerableSet by OZ to be associated storage.

this can be iterate on uninstall.

I also removed the InstallSession struct and collapsed everything into the EnableSession struct for better readability.

also changed the way how the sessionId is derived. check:

```
    function getSignerId(ISigner isigner, bytes memory isignerInitData) public pure returns (SignerId signerId);
```

as discussed with @filmakarov it might be good to add a salt to the session creation

### CODE REVIEW

File: contracts/DataTypes.sol [hunk 1]
```diff
@@ -39,15 +39,6 @@ struct SignerConf {
     FlatBytesLib.Bytes config;
 }
 
-// InstallSessions[] sessions;
-
-struct InstallSessions {
```
Comment (zeroknots) [inline contracts/DataTypes.sol:44]: removed this entirely. now only relying on EnableSessions

File: contracts/SmartSessionBase.sol [hunk 2]
```diff
@@ -108,18 +148,32 @@ abstract contract SmartSessionBase is ERC7579ValidatorBase {
...
         enableSessions(sessions);
     }
 
     /**
      * De-initialize the module with the given data
-     *
-     * @param data The data to de-initialize the module with
      */
-    function onUninstall(bytes calldata data) external override { }
+    function onUninstall(bytes calldata /*data*/ ) external override {
```
Comment (zeroknots) [inline contracts/SmartSessionBase.sol:164]: uninstall flow using EnumerableSet

File: contracts/lib/PolicyLib.sol [hunk 3]
```diff
@@ -181,10 +181,9 @@ library PolicyLib {
             ActionId actionId = actionPolicyData.actionId;
             SessionId sessionId = signerId.toSessionId(actionId, smartAccount);
             if (
-                $self.enabledActionIds[signerId].contains(smartAccount, ActionId.unwrap(actionId))
-                    && $self.actionPolicies[actionId].areEnabled(
-                        signerId, sessionId, smartAccount, actionPolicyData.actionPolicies
-                    )
+                $self.actionPolicies[actionId].areEnabled(
```
Comment (zeroknots) [inline contracts/lib/PolicyLib.sol:184]: performance optimization. imo its not necessary to for loop the storage

File: contracts/utils/AssociatedArrayLib.sol [hunk 4]
```diff
@@ -0,0 +1,151 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.20;
+
+library AssociatedArrayLib {
```
Comment (zeroknots) [inline contracts/utils/AssociatedArrayLib.sol:4]: refactored this and added new features.

I removed the "contains" function. for this its better to use the EnumerableSet4337

**Comment by filmakarov** [comment]
> as discussed with @filmakarov it might be good to add a salt to the session creation
does it mean, we'll have to include a salt into `EnableSessions` struct?

**Comment by zeroknots** [comment]
> > as discussed with @filmakarov it might be good to add a salt to the session creation
> 
> does it mean, we'll have to include a salt into `EnableSessions` struct?

I think this would be good. will add this

**Pull request merged**